__tests__/pet.test.js

```diff
@@ -0,0 +1,125 @@
+const Pet = require('../src/pet');
+
+describe('constructor',() => {
+    it('returns an object', () => { 
+        expect(new Pet('Fido')).toBeInstanceOf(Object);
+    });
+});
+describe('returns a name from an instance',() => {       
+    it('returns the correct name when a new instance is created', () => {
+        const pet = new Pet('Fido');
+        expect(pet.name).toEqual('Fido');
+    });
+});
+describe('constructor', () => {
+    it('sets the name property', () => {
+      const petOne = new Pet('Fido');
+      expect(petOne.name).toEqual('Fido');
+    });
+});
+describe('constructor', () => {
+    it('has an initial age of 0', () => {
+      const petOne = new Pet('Fido');
+      expect(petOne.age).toEqual(0);
+    });
+});
+describe('growUp', () => {
+    it('increments the age by +1, hunger by +5 and fitness by -3', () => {
+          const petOne = new Pet('Fido');
+          petOne.growUp();
+          expect(petOne.age).toEqual(1);
+          expect(petOne.hunger).toEqual(5);
+          expect(petOne.fitness).toEqual(7);
+    });  
+});
+describe('walk', () => {
+    it('increases the fitness by +4', () => {
+          const petOne = new Pet('Fido');
+          petOne.fitness = 4;
+          petOne.walk();
+          expect(petOne.fitness).toEqual(8);
+    }); 
+});
+describe('feed', () => {
+        it('decreases the hunger by 3', () => {
+              const petOne = new Pet('Fido');
+              petOne.hunger = 6;
+              petOne.feed();
+              expect(petOne.hunger).toEqual(3);
+        });     
+}); 
+describe('walk', () => {
+    it('increases the fitness to a MAX of 10', () => {
+        const petOne = new Pet('Fido');
+        petOne.fitness = 8;
+        petOne.walk();
+        expect(petOne.fitness).toEqual(10);
+    }); 
+});
+describe('constructor', () => {
+    it('has an initial hunger of 0', () => {
+      const petOne = new Pet('Fido');
+      expect(petOne.hunger).toEqual(0);
+    });
+});
+describe('constructor', () => {
+    it('has an initial fitness of 10', () => {
+      const petOne = new Pet('Fido');
+      expect(petOne.fitness).toEqual(10);
+    });
+});
+describe('checkUp status', () => {
+    it('checks the health of the pet', () => {
+        const petOne = new Pet('Fido');
+        petOne.fitness = 0;
+        petOne.hunger = 1;
+        expect(petOne.fitness <= 3 && petOne.hunger <=5).toBe(true);
```

Comment:
These assertions dont make a lot of sense, they will pass but it doesnt really tell us about the status of the pet. We have the checkup function so we should use it to assert the value returned is what it should be in each of these scenarios, for example the first one here could be replaced with:
```
        expect(petOne.checkUp()).toBe('I need a walk!')
```
Hardcoding the values in like this isnt the best practice, an alternative would be to have a separate file containing all the constants and error messages like this so you could reference them instead of writing them all out manually like this. Just food for thought for future development :)